testsuite/testcases/src/modifiers.rs

```diff
@@ -0,0 +1,202 @@
+use rand::Rng;
+use tokio::runtime::Runtime;
+
+fn add_execution_delay(swarm: &mut dyn Swarm, config: &ExecutionDelayConfig) -> anyhow::Result<()> {
+    let runtime = Runtime::new().unwrap();
+    let validators = swarm.get_validator_clients_with_names();
+
+    runtime.block_on(async {
+        let mut rng = rand::thread_rng();
+        for (name, validator) in validators {
+            let sleep_percentage = if rng.gen_bool(config.inject_delay_node_fraction) {
+                rng.gen_range(1_u32, config.inject_delay_max_transaction_percentage)
+            } else {
+                0
+            };
+            let name = name.clone();
+            info!(
+                "Validator {} adding {}% of transactions with {}ms execution delay",
+                name, sleep_percentage, config.inject_delay_per_transaction_ms
+            );
+            validator
+                .set_failpoint(
+                    "aptos_vm::execution::user_transaction".to_string(),
+                    format!(
+                        "{}%delay({})",
+                        sleep_percentage, config.inject_delay_per_transaction_ms
+                    ),
+                )
+                .await
+                .map_err(|e| {
+                    anyhow::anyhow!(
+                        "set_failpoint to add execution delay on {} failed, {:?}",
+                        name,
+                        e
+                    )
+                })?;
+        }
+        Ok::<(), anyhow::Error>(())
+    })
+}
+
+fn remove_execution_delay(swarm: &mut dyn Swarm) -> anyhow::Result<()> {
+    let runtime = Runtime::new().unwrap();
+    let validators = swarm.get_validator_clients_with_names();
+
+    runtime.block_on(async {
+        for (name, validator) in validators {
+            let name = name.clone();
+
+            validator
+                .set_failpoint(
+                    "aptos_vm::execution::block_metadata".to_string(),
+                    "off".to_string(),
+                )
+                .await
+                .map_err(|e| {
+                    anyhow::anyhow!(
+                        "set_failpoint to remove execution delay on {} failed, {:?}",
+                        name,
+                        e
+                    )
+                })?;
+        }
+        Ok::<(), anyhow::Error>(())
+    })
+}
+
+/// Config for additing variable processing overhead/delay into
+/// execution, to make different nodes have different processing speed.
+pub struct ExecutionDelayConfig {
+    /// Fraction (0.0 - 1.0) of nodes on which any delay will be introduced
+    pub inject_delay_node_fraction: f64,
+    /// For nodes with delay, what percentage (0-100) of transaction will be delayed.
+    /// (this is needed because delay that can be introduced is integer number of ms)
+    /// Different node speed come from this setting, each node is selected a number
+    /// between 1 and given max.
+    pub inject_delay_max_transaction_percentage: u32,
```

Comment:
Can you clarify why is this needed? Why not varying `inject_delay_per_transaction_ms` per node is not sufficient?